src/main/resources/net/uaznia/lukanus/hudson/plugins/gitparameter/javascript/git-parameter-select.js

```diff
@@ -11,67 +11,72 @@ function gitParameterUpdateSelect(listBox, url, divId, config) {
+            // if no value is explicitly selected by the server, try to select the same value
+            if (!selectionSet && possibleIndex != null)
+                l.selectedIndex = possibleIndex;
 
-        if (originalOnSuccess !== undefined)
-            originalOnSuccess(rsp);
+            if (originalOnSuccess !== undefined)
+                originalOnSuccess(rsp);
 
-        var errors = eval('(' + rsp.responseText + ')').errors
-        let error_div = $("git_parameter_errors_" + divId);
-        if (errors.length !== 0) {
-            error_div.show();
-            error_div.addClassName("error");
-            let $error_ul = $("git_parameter_errors_ul_" + divId);
-            var lis = "";
-            for (var j = 0; j < errors.length; j++) {
-                lis += "<li>" + escapeHTML(errors[j]) + "</li>"
+            var errors = json.errors
+            let error_div = $("git_parameter_errors_" + divId);
+            if (errors.length !== 0) {
+                error_div.style.display = "";
+                error_div.classList.add("error");
+                let $error_ul = $("git_parameter_errors_ul_" + divId);
+                var lis = "";
+                for (var j = 0; j < errors.length; j++) {
+                    lis += "<li>" + escapeHTML(errors[j]) + "</li>"
+                }
+                $error_ul.update(lis);
+            } else {
+                error_div.style.display = "none";
             }
-            $error_ul.update(lis);
-        }
-        else {
-            error_div.hide()
-        }
-
+        });
     };
     config.onFailure = function (rsp) {
-        l.removeClassName("select-ajax-pending");
-        status.innerHTML = rsp.responseText;
-        if (status.firstChild) {
-            status.firstChild.setAttribute('data-select-ajax-error', 'true')
-        }
-        Behaviour.applySubtree(status);
-        // deleting values can result in the data loss, so let's not do that unless instructed
-        var header = rsp.getResponseHeader('X-Jenkins-Select-Error');
-        if (header && "clear" === header.toLowerCase()) {
-            // clear the contents
-            while (l.length > 0) l.options[0] = null;
-        }
+        rsp.text().then(text => {
+            l.classList.remove("select-ajax-pending");
+            status.innerHTML = text;
+            if (status.firstChild) {
+                status.firstChild.setAttribute('data-select-ajax-error', 'true')
+            }
+            Behaviour.applySubtree(status);
+            // deleting values can result in the data loss, so let's not do that unless instructed
+            var header = rsp.headers.get('X-Jenkins-Select-Error');
+            if (header && "clear" === header.toLowerCase()) {
+                // clear the contents
+                while (l.length > 0) l.options[0] = null;
+            }
+        });
 
     };
 
-    l.addClassName("select-ajax-pending");
-    new Ajax.Request(url, config);
```

Comment:
The default in Prototype is POST, while the default in the Fetch API is GET, so this is technically a change in behavior. Since POST vs GET has security implications, the safer option is to avoid any change in behavior to avoid the risk of introducing a security regression.

Additionally, [this blog post](https://www.jenkins.io/blog/2023/05/12/removing-prototype-from-jenkins/) describes another gotcha: Jenkins core's patched version of Prototype adds a CSRF crumb to POST requests, but this needs to be done explicitly for the Fetch API.

Finally, the refactored code in this PR is not identical to the original code if, say, a 404 is returned from the server. In case the promise will not be rejected and we will execute the `config.onSuccess` callback in the new code, while the original code would have invoked the `config.onFailure` callback. To make the new code equivalent to the old code, we need to check `response.ok`. Something like this:

```javascript
fetch(url, {
  method: "POST",
  headers: crumb.wrap({}),
}).then((response) => {
  if (response.ok) {
    config.onSuccess(response);
  } else {
    config.onFailure(response);
  }
});
```